ecoscope/io/earthranger.py

```diff
@@ -178,11 +179,39 @@ def get_subjects(
             except IndexError:
                 raise KeyError("`group_name` not found")
 
-        df = pd.DataFrame(
-            self.get_objects_multithreaded(
-                object="subjects/", threads=self.tcp_limit, page_size=self.sub_page_size, **params
+        chunk_requests = False
+        if params.get("id") is not None:
+            subject_ids = params.get("id").split(",")
+
+            if len(subject_ids) > max_ids_per_request:
+                chunk_requests = True
+
+        if chunk_requests:
+            request_ids = [
+                subject_ids[i : i + max_ids_per_request] for i in range(0, len(subject_ids), max_ids_per_request)
+            ]
+            subjects = []
+
+            for chunked_ids in request_ids:
+                chunked_params = params.copy()
+                chunked_params["id"] = ",".join(chunked_ids)
+                subjects.append(
+                    pd.DataFrame(
+                        self.get_objects_multithreaded(
+                            object="subjects/", threads=self.tcp_limit, page_size=self.sub_page_size, **chunked_params
+                        )
+                    )
+                )
+
+            df = pd.concat(subjects, ignore_index=True)
+
+        else:
+            df = pd.DataFrame(
+                self.get_objects_multithreaded(
+                    object="subjects/", threads=self.tcp_limit, page_size=self.sub_page_size, **params
+                )
             )
```

Comment:
What about something a bit more succinct where we don't have to test for 'chunking' and we automatically do it each time. Does this work: 

```python
max_ids_per_request=50

def partial_subjects(x):
  params["id"] = x
  pd.DataFrame(self.get_objects_multithreaded(object="subjects/", threads=self.tcp_limit, page_size=self.sub_page_size, **params))

pd.concat([partial_subjects(x) for x in np.array_split(params["id"], len(params["id"])//max_ids_per_request)])
```

Reply:
I'm certainly not in love with the current structure, but I didn't even consider doing it this way! I'll test this out